### PR TITLE
Finish report api, listsurvey count and test case.

### PR DESCRIPTION
**Pull request by phejimlin**
Please review my code!

### CODE REVIEW

File: api/feedback/feedback.js [hunk 1]
```diff
@@ -149,6 +150,185 @@ module.exports = (() => {
    * Parameters:
    * Key        Description
    * surveyid   an uuid is generated by backend and associate to a survey as it creates first time.
+   * startKey   If your query amount to more than 1 MB of data, you'll need to perform another query request for the next 1 MB of data.
```
Comment (jonascheng) [inline api/feedback/feedback.js]: This should be removed, right?

File: api/feedback/feedback.js [hunk 2]
```diff
@@ -149,6 +150,185 @@ module.exports = (() => {
...
+    else {
+      return callback(new Error("400 Bad Request: Missing parameters: " + JSON.stringify(event)), null);
+    }
+  };
+
+  /**
+   * Parameters:
+   * Key        Description
+   * surveyid   an uuid is generated by backend and associate to a survey as it creates first time.
+   * startKey   If your query amount to more than 1 MB of data, you'll need to perform another query request for the next 1 MB of data.
```
Comment (jonascheng) [inline api/feedback/feedback.js]: This should be removed, right?

File: api/report/s-function.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "report",
+  "runtime": "nodejs4.3",
+  "description": "${project} - ${stage}: ${name}",
+  "customName": "$${functionName}",
+  "customRole": false,
+  "handler": "report/handler.handler",
+  "timeout": 60,
```
Comment (jonascheng) [inline api/report/s-function.json]: Please set to max 300 in next PR.

File: api/report/s-function.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        }
+      },
+      "responses": {
+        "default": {
+          "statusCode": "200",
+          "responseParameters": {
+            "method.response.header.Content-disposition": "'attachment; filename=Qustom.csv'"
+          },
+          "responseTemplates": {
+            "text/csv": "$input.body.replaceAll(\"\"\"\",\"\").replaceAll(\"\\\u0027\",\"\"\"\").replaceAll(\"\\\\n\",\"\n\")"
```
Comment (jonascheng) [inline api/report/s-function.json]: This is very critical to document in TOI.

**Comment by jonascheng** [comment]
Jim,

  Sorry, I modified package.json before merging your PR.
  Would you please resolve it and re-submit PR?

Thanks,
Jonas

**Comment by jonascheng** [comment]
Sorry, also s-resources-cf.json need to be resolved.

File: api/feedback/feedback.js (hunk 1 above)
Comment (phejimlin) [inline api/feedback/feedback.js]: Yes! I forget to remove it.

File: api/report/s-function.json (hunk 3 above)
Comment (phejimlin) [inline api/report/s-function.json]: ok!

**Comment by jonascheng** [comment]
Thanks Jim! I will check CI status tomorrow, it keep failing in coveralls step.

It's very strange failure.

**Comment by phejimlin** [comment]
So weird~

**Pull request merged**